blackbox/docs/sql/administration/user_defined_functions.txt

```diff
@@ -232,10 +232,72 @@ the JavaScript function should return a ``GeoJson`` object or``WKT`` string::
 
 .. NOTE::
 
-   If the return value of the JavaScript function is ``undefined`` it is converted
-   to ``NULL``.
+   If the return value of the JavaScript function is ``undefined`` it is
+   converted to ``NULL``.
 
 
+Working with ``TIMESTAMP``
+..........................
+
+`Nashorn`_, the JavaScript interpreter that CrateDB uses, interprets numbers as
+``java.lang.Double``, ``java.lang.Long``, or ``java.lang.Integer``,
+depending on the computation performed. In most cases, this is not an
+issue, since the return type of the JavaScript function will be casted
+to the return type specified in the ``CREATE FUNCTION`` statement,
+although cast might result in a loss of precision.
+
+However, when you try to cast ``DOUBLE`` to ``TIMESTAMP`` it will be
+interpreted as UTC seconds and will result in a wrong value::
+
+     cr> CREATE FUNCTION utc(long, long, long)
+     ...  RETURNS TIMESTAMP
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function utc(year, month, day) {
+     ...        return Date.UTC(year, month, day, 0, 0, 0);
+     ...      }';
+     CREATE OK, 1 row affected  (... sec)
+
+.. hide:
+
+    cr> _wait_for_function('utc')
+
+
+::
+
+    cr> SELECT date_format(utc(2016,04,6)) as epoque;
+    +------------------------------+
+    | epoque                       |
+    +------------------------------+
+    | 48314-07-22T00:00:00.000000Z |
+    +------------------------------+
+    SELECT 1 row in set (... sec)
+
+To avoid this behavior, the numeric value should be divided by 1000 before
+it is returned::
```

Comment:
To avoid this behaviour, return a timestamp in seconds as floating point number. You can reach that, by dividing the timestamp with 1000.0